packages/backend/src/setup/modules.ts

```diff
@@ -0,0 +1,98 @@
+import Router from '@koa/router'
+import { HttpClient, Logger } from '@l2beat/common'
+
+import { ActivityController } from '../api/controllers/ActivityController'
+import { createActivityRouter } from '../api/routers/ActivityRouter'
+import { Config } from '../config'
+import { Clock } from '../core/Clock'
+import { ZksyncTransactionUpdater } from '../core/transaction-count/ZksyncTransactionUpdater'
+import { RpcTransactionCountRepository } from '../peripherals/database/RpcTransactionCountRepository'
+import { Database } from '../peripherals/database/shared/Database'
+import { StarkexTransactionCountRepository } from '../peripherals/database/StarkexTransactionCountRepository'
+import { ZksyncTransactionRepository } from '../peripherals/database/ZksyncTransactionRepository'
+import { StarkexClient } from '../peripherals/starkex'
+import { ZksyncClient } from '../peripherals/zksync'
+import { createRpcTransactionUpdaters } from './createRpcTransactionUpdaters'
+import { createStarkexTransactionUpdaters } from './createStarkexTransactionUpdaters'
+
+interface AppModule {
+  start?: () => void
+  router?: Router
+}
```

Comment:
This type is unnecessary and potentially does more harm then good. IMO `getActivityModule` should just return its own object or undefined without trying to generalize.

Also, I think that instead of this file being named `modules.ts` it should be named `ActivityModule.ts`

Reply:
I agree with @sz-piotr with one addition - imo instead of returning `undefined` when `transactionCountSync` is missing in `config`, `getActivityModule` should accept just `transactionCountSync` instead of full `config`, because it does not make sense to create this module otherwise - caller should be handling this.

Reply:
The problem with this is that its dependencies require full configs.

Reply:
Ah, you're right. IMO that's an issue with its dependencies (because they in fact do not require full config), but that's a bigger refactor and can be done in a separate PR.

Reply:
I think sometimes they actually do require parts of the config outside of counting related stuff. I think generally is simpler to treat config as a blob.